src/coreclr/nativeaot/Runtime/inc/stressLog.h

```diff
@@ -488,36 +488,89 @@ class StressLog {
 #endif
 
 //==========================================================================================
-// StressMsg
-//
 // The order of fields is important.  Keep the prefix length as the first field.
-// And make sure the timeStamp field is naturally aligned, so we don't waste
+// And make sure the timeStamp field is naturally alligned, so we don't waste
 // space on 32-bit platforms
-//
-struct StressMsg {
-    static const size_t formatOffsetBits = 26;
-    union {
-        struct {
-            uint32_t numberOfArgs  : 3;   // at most 7 arguments
-            uint32_t formatOffset  : formatOffsetBits;    // offset of string in mscorwks
-            uint32_t numberOfArgsX : 3;                   // extend number of args in a backward compat way
+struct StressMsg
+{
+private:
+    static const size_t formatOffsetLowBits = 26;
+    static const size_t formatOffsetHighBits = 13;
+
+    union
+    {
+        struct
+        {
+            // We split the format offset to ensure that we utilize every bit and that
+            // the compiler does not align the format offset to a new 64-bit boundary.
+            uint64_t facility: 32;                           // facility used to log the entry
+            uint64_t numberOfArgs : 6;                       // number of arguments
+            uint64_t formatOffsetLow: formatOffsetLowBits;   // offset of format string in modules
+            uint64_t formatOffsetHigh: formatOffsetHighBits; // offset of format string in modules
+            uint64_t timeStamp: 51;                          // time when msg was logged (100ns ticks since runtime start)
+        };
+
+        struct
+        {
+            uint64_t argsFacilityOffsetLow;
+            uint64_t timeStampOffsetHigh;
```

Comment:
```suggestion
    struct
    {
        // We split the format offset to ensure that we utilize every bit and that
        // the compiler does not align the format offset to a new 64-bit boundary.
         uint64_t facility: 32;                           // facility used to log the entry
         uint64_t numberOfArgs : 6;                       // number of arguments
         uint64_t formatOffsetLow: formatOffsetLowBits;   // offset of format string in modules
         uint64_t formatOffsetHigh: formatOffsetHighBits; // offset of format string in modules
         uint64_t timeStamp: 51;                          // time when msg was logged (100ns ticks since runtime start)
```
Can this be just this? 

The union does not seem to be actually useful.

Reply:
The union is required for the DebugHeader table (you can't do offsetof for bitfields)

Reply:
DebugHeader table describes the runtime data structure contracts. It works well for data structures that are composed from simple fields. It does not work well for bit fields.

We should not need offsets of these fields in the DebugHeader table since they are very imprecise descriptor of the contract.

Reply:
If the debug header fields aren’t useful, I’ll remove the fields from the debug header and remove the union.

Reply:
Thanks. It will need a matching fix in the data contract reader (cc @mikem8361).

Reply:
I can set up the fix there as well if someone sends me the link to it on Teams.

Reply:
Updated the PR to remove the union.